### PR TITLE
feat(EMS-3985-4004): broker addresses - dynamic intro, search again link

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR updates the "Broker addresses" page to:
- Render intro text with e.g "1 address found ...", "2 addresses found ..."
- Render a "Search again" link.

## Resolution :heavy_check_mark:
- Update page content strings.
- Update E2E test.
- Update `pageVariables` to return `INTRO` strings and `SEARCH_AGAIN_URL`.
- Update the GET controller to pass a `buildingNumberOrName` variable.
- Update nunjucks template.

## Miscellaneous :heavy_plus_sign:
List any additional fixes or improvements.

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    it('renders a intro copy', () => {
+      intro()
+        .invoke('text')
+        .then((text) => {
+          expect(text.trim()).includes(`1 ${ADDRESS} ${FOUND_FOR} `);
+          expect(text.trim()).includes('W1A 1AA');
+          expect(text.trim()).includes(` ${SEPARATOR} `);
+          expect(text.trim()).includes('WOGAN HOUSE.');
+          expect(text.trim()).includes(SEARCH_AGAIN);
+        });
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js:71]: no way to use assertText here?

File: e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js:1]: Is there an test for multiple addresses found?

File: e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js:71]: No, as it has child elements in the paragraph

File: e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/policy/broker-addresses/broker-addresses.spec.js:1]: No, that will be tackled in EMS-3991. This PR is purely for adding the intro text and "search again" link.

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**